recipes/h5pp/all/conanfile.py

```diff
@@ -0,0 +1,77 @@
+import os
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class h5ppConan(ConanFile):
+    name = "h5pp"
+    description = "A C++17 wrapper for HDF5 with focus on simplicity"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/DavidAce/h5pp"
+    topics = ("conan","h5pp","hdf5", "binary", "storage")
+    license = "MIT"
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package"
+    requires = "eigen/3.3.7", "spdlog/1.7.0", "hdf5/1.12.0"
+    exports_sources = ["CMakeLists.txt"]
```

Comment:
conanfile claims that a CMakeLists.txt is exported, but there is no such file in the recipe.
This CMakeLists wrapper allows to properly inject conan profile. See https://docs.conan.io/en/latest/reference/generators/cmake.html#conan-basic-setup

This is probably why you have linkage errors in test_package.

Reply:
I'm not quite sure I have understood this. So, is `exports_sources = ["CMakeLists.txt"]` an option to provide a `CMakeLists.txt` different from the one already in h5pp? How does that allow profile injection? I don't see the connection to the link (this is all still new to me). 

In any case, studying the log I saw the the undefined references for `std::filesystem` that are expected when -lstdc++fs is missing on some compilers. The cmake configuration in h5pp should be able to add the appropriate link flags when needed but I think the CI must have found an edge case.

Reply:
Yes you're right, probably comes from stdc++fs. But it's better to add CMakeLists wrapper, even for header_only library. See in this recipe for example: https://github.com/conan-io/conan-center-index/blob/master/recipes/cereal/all/CMakeLists.txt

Reply:
You can also get rid of CMake in this recipe and just copy headers folder (in `package` method).

Reply:
Aha okay, got it. Well I think the cmake from h5pp currently is almost equivalent to the wrapper (see [here](https://github.com/DavidAce/h5pp/blob/59d7da7b6e9d421a4185d69b84be6d9732a81f12/cmake/SetupDependenciesConan.cmake#L5)). I'll keep this in mind.

Reply:
correct, therefore this `exports_sources` is useless
```suggestion
```